tools/setup_generation/step_refman.py

```diff
@@ -397,34 +399,31 @@ def generate_entries(
             xrefs_output_file.write(json.dumps(xrefs))
 
     @staticmethod
-    def add_external_methods_to_config_class():
-        import os
+    def add_external_methods_to_config_class(setup: Setup):
         if not os.path.exists("config_doc.txt"):
-            print(f"WARNING - No method found to add to Config documentation !")
+            print(f"WARNING - No methods found to inject to Config documentation!")
             return
 
         # Get code of methods to inject
-        with open('config_doc.txt', 'r') as f:
-            print(f"INFO - Found some methods to add to Config documentation.")
+        with open("config_doc.txt", "r") as f:
+            print(f"INFO - Injecting methods to Config documentation.")
             methods_to_inject = f.read()
 
         # Delete temporary file
         if os.path.exists("config_doc.txt"):
             os.remove("config_doc.txt")
 
-        # Copy config.py file
-        from pathlib import Path
-        path = Path("tools", "taipy", "config", "config.py")
-        back_up_path = Path("tools", "setup_generation", "config.py.bak")
-        shutil.copyfile(path, back_up_path)
+        # Backup file taipy/config/config.py
+        taipy_config_dir = os.path.join(setup.tools_dir, "taipy", "config")
+        config_path = os.path.join(taipy_config_dir, "config.py")
+        shutil.copyfile(config_path, os.path.join(taipy_config_dir, "config.py.bak"))
 
         # Read config.py file
-        with open(path, 'r') as f:
+        with open(config_path, "r") as f:
             contents = f.readlines()
 
         # Inject imports and code
-        imports_to_inject = """from types import NoneType
-from typing import Any, Callable, List
+        imports_to_inject = """from typing import Any, Callable, List
```

Comment:
you removed NoneType? 
Is it on purpose?

Reply:
Well yes: it is not used... and it actually broke my setup at times...
Go figure.